addons/sale_mrp/models/sale_mrp.py

```diff
@@ -41,7 +41,7 @@ def _compute_qty_delivered(self):
                     # the products for this PO will set the qty_delivered. We might need to check the
                     # state of all PO as well... but sale_mrp doesn't depend on purchase.
                     if dropship:
-                        if order_line.move_ids and all([m.state == 'done' for m in order_line.move_ids]):
+                        if order_line.move_ids and all([m.state == 'done' for m in order_line.move_ids.filtered(lambda m: m.state != 'cancel')]):
```

Comment:
The patch is good, but I would write 
```python
all([m.state in ('done', 'cancel') for m in order_line.move_ids])
``` 
instead.

Reply:
what about every moves are cancelled ?

Reply:
If it is a dropship and there is a matching kit BoM, I would say to follow the usual flow and use the method _compute_kit_quantities with the moves that are done. Some context could be passed to the method so that the bom_line_moves are all the ones being passed, instead of filtering like:

`bom_line_moves = self.filtered(lambda m: m.bom_line_id == bom_line)`

Reply:
It would require some refactoring however..., what about simply checking that in the done move_ids there are all the components? We have the BoM, we can use the explode.

Reply:
I guess another option is to check that not all moves are cancelled, and keep the fix